### PR TITLE
Peggy's todo app - Remember

### PR DESCRIPTION
**Pull request by pwangy**
Link to my deployed project: https://pwangy-todo.netlify.app/

My reflections on how this week's project turned out: 
I was able to hit the blue level goals and that feels like a an OK stopping off point for now. 

This is definitely a project that I'm planning to come back around and tinker with... 
- finish sorting out a "clear all done" button
- try out a unique-id node package
- local storage
- create an empty state

After being focused on getting some basic functionality done, I started in on styling a bit to get this app demo-ready. Then decided to try looking into grids and ended up getting sidetracked trying to convert my layout to being grid based and learning about that (that work is in a yet unmerged branch.) It still feels unfinished on my part, however, I wanted to turn in what I had and come back around to it again. 

Redux is fun and I like it even though it's something I'm still picking up. 

I'm still struggling with figuring out a good system of how to use Styled-Components where I want that styling to live. I feel conflicted because theoretically one-off styling should likely live in the .js file where it's being used, but it seems cleaner to keep it in a separate component on it's own... except that you end up with a zillion components which seems like it could get out of hand quickly. This week I ended up creating components I meant to be more generic (till example: one generic button style stored in 'lib') but found that something like a container, while being a basic item often needed to become custom in order to specify where in the grid layout it should appear. I think it'll all get clearer with more practice so I hope that eventually I'll get a handle on it and it'll stop feeling like little bits of styling are living all over the place. I enjoy modularity, but it can be hard to switch over to thinking this way.

Things I'd like to have clarified or explained in more detail: Local storage, please.

### CODE REVIEW

File: code/src/components/ClearButton.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import { remember } from 'reducers/remember'
+import { Button } from 'lib/Button'
+
+export const ClearButton = () => {
+  const dispatch = useDispatch()
+
+  return (
+    <Button
+      background="salmon"
```
Comment (IngelaL) [inline code/src/components/ClearButton.js:12]: Really smart that you used reusable styled components for the buttons and that you use props to change color for some of them. 👏

File: code/src/components/AddTask.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            onChange={(event) => setPriority(event.target.value)}>
+            <option value="">Select...</option>
+            <option value="High">High</option>
+            <option value="Medium">Medium</option>
+            <option value="Low">Low</option>
+          </Select>
+        </label>
+        <Button type="submit">Add Item</Button>
+      </form>
+    </RoundedContainer>
```
Comment (IngelaL) [inline code/src/components/AddTask.js:45]: Clean and well strucured code. Looks great. (it goes for all of your code) ✨

File: code/src/components/List.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <RoundedContainer>
+        <label>
+          Filter by:
+          <Select
+            value={priority}
+            onChange={(event) => setPriority(event.target.value)}>
+            <option value="High">High</option>
+            <option value="Medium">Medium</option>
+            <option value="Low">Low</option>
+            <option value="">Show All</option>
```
Comment (IngelaL) [inline code/src/components/List.js:30]: It is nice that you changed the color in dropdown lists! To make the text in the dropdown easier to read you could change it to a brighter color though.

File: code/src/components/AddTask.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        <label>
+          Set Priority:
+          <Select
+            value={priority}
+            onChange={(event) => setPriority(event.target.value)}>
+            <option value="">Select...</option>
+            <option value="High">High</option>
+            <option value="Medium">Medium</option>
+            <option value="Low">Low</option>
```
Comment (IngelaL) [inline code/src/components/AddTask.js:40]: Nice that you can choose priority. Very practical and a great way to sort tasks. 👌

File: code/src/reducers/remember.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      state.items = state.items.filter((item) => item.id !== action.payload)
+    },
+    clearDone: (state, action) => {
+      state.items = state.items.filter((item) => item.id.done === action.payload)
+    },
+    clearAll: (state, action) => {
+      state.items = []
+    },
+    toggleDone: (state, action) => {
+      console.log(action.payload)
```
Comment (IngelaL) [inline code/src/reducers/remember.js:30]: Dont forget to delete console.logs!

File: README.md [hunk 6]
```diff
@@ -1,13 +1,47 @@
...
+- React
+- React Hooks
+- JSX -->
+
+
+
+## Requirements Completed
+🔵  Blue Level
+- Your app should list all todo tasks - completed or uncompleted
+- You should be able to mark an uncompleted task as complete (and change it back to uncomplete)
```
Comment (IngelaL) [inline README.md]: Don't forget to replace the text in the read-me file.

File: code/public/index.html [hunk 7]
```diff
@@ -4,31 +4,13 @@
...
-      Notice the use of %PUBLIC_URL% in the tags above.
-      It will be replaced with the URL of the `public` folder during the build.
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>Technigo React App</title>
+    <link rel="icon" href="assets/favicon.png" type="image/png">
```
Comment (IngelaL) [inline code/public/index.html:7]: I learned something new here 😊 Looks very nice with an icon next to the title name.

File: code/src/components/AddTask.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
```
Comment (IngelaL) [inline code/src/components/AddTask.js:1]: Should this be deleted or what is it for?

File: code/src/components/CustomCheckbox.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable no-shadow */
```
Comment (IngelaL) [inline code/src/components/CustomCheckbox.js:2]: Curious about these :) Delete?

File: code/src/components/CustomCheckbox.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  const Checkbox = ({ className, checked, ...props }) => (
+    <CheckboxContainer>
+      <HiddenCheckbox checked={checked} {...props} />
+      <StyledCheckbox checked={checked}>
+        <Icon viewBox="0 0 24 24">
+          <polyline points="20 6 9 17 4 12" />
+        </Icon>
+      </StyledCheckbox>
+    </CheckboxContainer>
+  )
```
Comment (IngelaL) [inline code/src/components/CustomCheckbox.js:60]: Cool checkboxes and the way you styled them!

File: code/src/App.js [hunk 11]
```diff
@@ -1,9 +1,28 @@
...
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
+    <Provider store={store}>
+      <AppShell>
+        <Summary />
+        <AddTask />
+        <List />
+        {/* <Footer /> */}
```
Comment (IngelaL) [inline code/src/App.js:24]: Footer not added but I guess that is on your todo-list :)

File: code/src/components/AddTask.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          value={text}
+          onChange={(event) => setText(event.target.value)}
+          placeholder="add an item"
+          required />
+
+        <label>
+          Set Priority:
```
Comment (IngelaL) [inline code/src/components/AddTask.js:33]: "Set priority" needs a bit margin to the left so it is not so close to the input field.

**Review by IngelaL — APPROVED** [review]
Hi Peggy!
I think that you have made a user friendly and practical todo-list. The design is very nice and you have used styled components in a great way by making them reusable. Also, cool that you used props to change color of buttons and divs. 

Don't forget to add your own text to the read-me file and add the footer that you probably didn't have time to finish. I can see that you put a lot of time in this project and great that you have plans of adding even more functionalities. Great job!

File: code/src/App.js (hunk 11 above)
Comment (EmOlsz) [reply code/src/App.js:24]: If not, let's remove it ;)

File: code/src/components/AddTask.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <RoundedContainer>
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          value={text}
+          onChange={(event) => setText(event.target.value)}
+          placeholder="add an item"
+          required />
```
Comment (EmOlsz) [inline code/src/components/AddTask.js:30]: I'm happy you remembered about not accepting empty todos!

File: code/src/components/ClearDoneButton.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+export const ClearDone = () => {
+  const dispatch = useDispatch()
+
+  return (
+    <Button
+      background="salmon"
+      onClick={() => {
+        dispatch(remember.actions.clearDone())
+      }}>
```
Comment (EmOlsz) [inline code/src/components/ClearDoneButton.js:15]: There is no need for this extra set of curly brackets, this is completely enough :

```
onClick={() => dispatch(remember.actions.clearDone())}>
```

File: code/src/components/Summary.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+import { Container, RoundedContainer } from 'lib/Container'
+import { Icon } from 'lib/Icon'
+import { Title, Subtitle, H3 } from 'lib/Text'
+
+export const Summary = () => {
+  const items = useSelector((state) => state.remember.items)
+  const toDo = items.filter((item) => item.done !== true)
```
Comment (EmOlsz) [inline code/src/components/Summary.js:10]: Or just :

```
const toDo = items.filter((item) => !item.done)
```

**Pull request closed without merging**